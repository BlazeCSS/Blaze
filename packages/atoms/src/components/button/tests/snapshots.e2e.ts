import snapIt from '../../../test/snap-it';

const component = 'button';

describe(component, () => {
  const snap = snapIt(component);

  describe('renders', () => {
    snap('a default button', `<blaze-button>Action</blaze-button>`);
    snap('an a button', `<blaze-button href="nowhere.com" value="Action"></blaze-button>`);
    snap('a colored button', `<blaze-button type="info">Action</blaze-button>`);
    snap('a sized button', `<blaze-button size="small">Action</blaze-button>`);
    snap('a full-width button', `<blaze-button full>Action</blaze-button>`);
    snap('a ghost button', `<blaze-button ghost>Action</blaze-button>`);
    snap('a colored ghost button', `<blaze-button ghost type="info">Action</blaze-button>`);
    snap('a rounded button', `<blaze-button rounded>Action</blaze-button>`);
    snap('a rounded ghost button', `<blaze-button rounded ghost>Action</blaze-button>`);
    snap('an active button', `<blaze-button active>Action</blaze-button>`);
  });
});
