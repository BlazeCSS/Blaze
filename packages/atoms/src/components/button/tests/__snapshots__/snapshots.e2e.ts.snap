// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders a colored button 1`] = `"<blaze-button type=\\"info\\" class=\\"hydrated\\"><button class=\\"c-button c-button--info\\">Action</button></blaze-button>"`;

exports[`button renders a colored ghost button 1`] = `"<blaze-button ghost=\\"\\" type=\\"info\\" class=\\"hydrated\\"><button class=\\"c-button c-button--ghost-info\\">Action</button></blaze-button>"`;

exports[`button renders a default button 1`] = `"<blaze-button class=\\"hydrated\\"><button class=\\"c-button\\">Action</button></blaze-button>"`;

exports[`button renders a full-width button 1`] = `"<blaze-button full=\\"\\" class=\\"hydrated\\"><button class=\\"c-button c-button--block\\">Action</button></blaze-button>"`;

exports[`button renders a ghost button 1`] = `"<blaze-button ghost=\\"\\" class=\\"hydrated\\"><button class=\\"c-button c-button--ghost\\">Action</button></blaze-button>"`;

exports[`button renders a rounded button 1`] = `"<blaze-button rounded=\\"\\" class=\\"hydrated\\"><button class=\\"c-button c-button--rounded\\">Action</button></blaze-button>"`;

exports[`button renders a rounded ghost button 1`] = `"<blaze-button rounded=\\"\\" ghost=\\"\\" class=\\"hydrated\\"><button class=\\"c-button c-button--ghost c-button--rounded\\">Action</button></blaze-button>"`;

exports[`button renders a sized button 1`] = `"<blaze-button size=\\"small\\" class=\\"hydrated\\"><button class=\\"c-button u-small\\">Action</button></blaze-button>"`;

exports[`button renders an a button 1`] = `"<blaze-button href=\\"nowhere.com\\" value=\\"Action\\" class=\\"hydrated\\"><a href=\\"nowhere.com\\" class=\\"c-button\\"></a></blaze-button>"`;

exports[`button renders an active button 1`] = `"<blaze-button active=\\"\\" class=\\"hydrated\\"><button class=\\"c-button c-button--active\\">Action</button></blaze-button>"`;
